packages/db/tests/models/test-collection-model.js

```diff
@@ -79,3 +82,144 @@ test('CollectionPgModel.upsert() overwrites a collection record', async (t) => {
     }
   );
 });
+
+test('Collection.searchWithUpdatedAtRange() returns an array of records if no date range specified', async (t) => {
+  const {
+    knex,
+    collectionPgModel,
+  } = t.context;
+
+  const collectionName = cryptoRandomString({ length: 10 });
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+  }));
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    { name: collectionName },
+    {}
+  );
+
+  t.is(searchResponse.length, 3);
+});
+
+test('Collection.searchWithUpdatedAtRange() returns a filtered array of records if a date range is specified', async (t) => {
+  const {
+    knex,
+    collectionPgModel,
+  } = t.context;
+
+  const collectionName = cryptoRandomString({ length: 10 });
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+  }));
+
+  const dateValue = 5000;
+  const searchRecord = fakeCollectionRecordFactory({
+    name: collectionName,
+    version: '4',
+    updated_at: new Date(dateValue),
+  });
+  records.push(searchRecord);
+
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    {
+      name: collectionName,
+    },
+    {
+      updatedAtFrom: new Date(dateValue - 1),
+      updatedAtTo: new Date(dateValue + 1),
+    }
+  );
+
+  t.is(searchResponse.length, 1);
+  t.like(
+    removeNilProperties(searchResponse[0]),
+    { ...searchRecord, ...{ files: JSON.parse(searchRecord.files) } }
+  );
+});
+
+test('Collection.searchWithUpdatedAtRange() returns a filtered array of records if only updatedAtTo is specified', async (t) => {
+  const {
+    knex,
+    collectionPgModel,
+  } = t.context;
+
+  const dateValue = 5000;
+  const collectionName = cryptoRandomString({ length: 10 });
+  const records = times(3, (i) => fakeCollectionRecordFactory({
+    name: collectionName,
+    version: i,
+  }));
+
+  const searchRecord = fakeCollectionRecordFactory({
+    name: collectionName,
+    updated_at: new Date(dateValue),
+    version: '4',
+  });
+  records.push(searchRecord);
+
+  await Promise.all(records.map((r) => collectionPgModel.create(knex, r)));
+
+  const searchResponse = await collectionPgModel.searchWithUpdatedAtRange(
+    knex,
+    {
+      name: collectionName,
+    },
+    {
+      updatedAtTo: new Date(dateValue + 1),
```

Comment:
So if only `updatedAtTo` is specified, then it should be returning all records from the beginning of time to the `updatedAtTo` time? And we know that will be only 1 record because all the records where we're not setting `dateValue` have a much later seconds from the epoch than `5000`? 

Would it be worth explicitly specifying the `updatedAt` for the other records to make this explicit?

Reply:
👍🏻  It's probably a good idea, given the Record Factory defaults could change: 5081a299d